pkg/cmd/api/api.go

```diff
@@ -178,6 +180,10 @@ func apiRun(opts *ApiOptions) error {
 		return err
 	}
 
+	if opts.Silent {
+		opts.IO.Out = ioutil.Discard
```

Comment:
This is a neat trick! But, since parsing response JSON will take some processing time and will ultimately be discarded, could you instead skip the body processing logic in "silent" mode and exit from the command early?

Reply:
On the other hand, I just realized that the response body needs to be first consumed in order for our response error processing logic to work. So, maybe discard my above comment please, as following my suggestion might lead to too many code changes. I'll leave it to you

Reply:
I think this Discard option is better than some complex logic. I agree with you on the error handling part. Even if I put the custom logic and all, it will become too bloated, and will cause problems down the line also, in case we need to add/change some feature/working.